### PR TITLE
Add a flag to indicate whether to run mixed precision according to resource utilization provided by user.

### PR DESCRIPTION
**Pull request by lior-dikstein**
## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py [hunk 1]
```diff
@@ -99,23 +99,21 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
...
 
     def get_mixed_precision_config(self):
         return mct.core.MixedPrecisionQuantizationConfig(num_of_images=1,
                                                          distance_weighting_method=self.distance_metric)
 
     def compare(self, quantized_model, float_model, input_x=None, quantization_info=None):
         conv_layers = get_layers_from_model_by_type(quantized_model, layers.Conv2D)
-        assert (quantization_info.mixed_precision_cfg == [0,
-                                                          0]).all()  # resource utilization is infinity -> should give best model - 8bits
+        assert (quantization_info.mixed_precision_cfg == self.expected_mp_config).all()
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py:116]: I know it's not your original code, but if you already touched it, change the assert to some unitTest assertion

File: tests/pytorch_tests/model_tests/feature_models/mixed_precision_activation_test.py [hunk 2]
```diff
@@ -119,12 +119,26 @@ def compare(self, quantized_models, float_model, input_x=None, quantization_info
 class MixedPercisionActivationMultipleInputs(MixedPercisionActivationBaseTest):
     def __init__(self, unit_test):
         super().__init__(unit_test)
-        self.expected_config = [0 for _ in range(8)]
+        self.expected_config = [0 for _ in range(8)] + [1]
```
Comment (ofirgo) [inline tests/pytorch_tests/model_tests/feature_models/mixed_precision_activation_test.py:122]: why the +[1]? maybe add comment cause it's not clear

**Review by ofirgo — CHANGES_REQUESTED** [review]
Some cosmetic comments.
In addition:
1. Please add a dedicated test for the new feature (define a setup with TPC that would run mixed precision before the new feature, and verify that it didn't run mixed precision). I would replicate this test for each of the available resource utilization types.
2. Please update the release notes draft.
3. Also add a description to the PR with an explanation about the feature (and include it in the commit message when you merge it).

File: model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_data.py [hunk 3]
```diff
@@ -78,20 +84,24 @@ def compute_resource_utilization_data(in_model: Any,
...
+    Computes an array of the respective weights memory in bytes and an array of the respective
+            number of weights parameters for each node.
 
     Args:
         graph: Finalized Graph object.
         fw_info: FrameworkInfo object about the specific framework
             (e.g., attributes of different layers' weights to quantize).
 
-    Returns: A vector of node's weights memory sizes.
+    Returns: An array of the graph's nodes weights memory in bytes and an array of the graph's
```
[the remark this answers is not in the document]
Comment (lior-dikstein) [reply model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_data.py:98]: yes

File: model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_data.py [hunk 4]
```diff
@@ -100,6 +110,11 @@ def compute_nodes_weights_params(graph: Graph, fw_info: FrameworkInfo) -> np.nda
         kernel_attr = fw_info.get_kernel_op_attributes(n.type)[0]
         if kernel_attr is not None and not n.reuse:
             kernel_candidates = n.get_all_weights_attr_candidates(kernel_attr)
+            weight_n_bit_candidates = [kc.weights_n_bits for kc in
+                                     kernel_candidates] if n.candidates_quantization_cfg is not None else []
+            max_weight_bits = max(weight_n_bit_candidates) if len(weight_n_bit_candidates) > 0 else 0
```
[the remark this answers is not in the document]
Comment (lior-dikstein) [reply model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_data.py:115]: Refactored the whole function to make it better and more readable

**Review by ofirgo — CHANGES_REQUESTED** [review]

File: tests/keras_tests/feature_networks_tests/feature_networks/mixed_precision/requires_mixed_precision_test.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/feature_networks/mixed_precision/requires_mixed_precision_test.py:1]: missing license

**Review by ofirgo — APPROVED** [review]
